option optimize_for = LITE_RUNTIME;

package Safir.Dob.Internal.SP;

message NodeStatisticsMessage
{
    message NodeInfo 
    {       
        //This info is about other nodes that this node talks to
        required string name = 1;
        required sfixed64 id = 2;
        required string control_address = 3; //The address and port of the node's control socket
        required bool multicast_enabled = 4;

        required bool is_dead = 5;
        required uint32 receive_count = 6;
        required uint32 retransmit_count = 8;

        //if this message is a node-local SP to SP message (e.g. between control and dose_main and dobexplorer)
        //it will contain all the node statistics received from other nodes.
        optional NodeStatisticsMessage remote_statistics = 9;
    }
    
    //TODO: how do we stop this message from growing infinitely when there are lots of
    //nodes that have started and stopped?
    
    //information about myself
    required string name = 1;
    required sfixed64 id = 2;
    required string control_address = 3; //The address of the node's control socket
    required string multicast_address = 4; //The address of the node's control socket
    
    repeated NodeInfo node_info = 5;

}


